services/web/client/source/class/qxapp/component/widget/FileManager.js

```diff
@@ -0,0 +1,319 @@
+        backgroundColor: "white",
+        minWidth: 300,
+        height: 24
+      });
+
+      let downloadBtn = new qx.ui.form.Button().set({
+        icon: "@FontAwesome5Solid/cloud-download-alt/24"
+      });
+      downloadBtn.addListener("execute", function(e) {
+        this.__downloadFile();
+      }, this);
+
+      let deleteBtn = new qx.ui.form.Button().set({
+        icon: "@FontAwesome5Solid/trash-alt/24"
+      });
+      deleteBtn.addListener("execute", function(e) {
+        this.__deleteFile();
+      }, this);
+
+      selectedFileLayout.add(selectedLabel, {
+        flex: 1
+      });
+      selectedFileLayout.add(downloadBtn);
+      selectedFileLayout.add(deleteBtn);
+    }
+
+    let closeBtn = this.__closeBtn = this._createChildControlImpl("closeButton");
+    closeBtn.setEnabled(false);
+    closeBtn.addListener("execute", function() {
+      console.log("close");
+    }, this);
+
+    this.__reloadNodeTree();
+    this.__reloadUserTree();
+  },
+
+  properties: {
+    nodeModel: {
+      check: "qxapp.data.model.NodeModel"
+    }
+  },
+
+  members: {
+    __nodeTree: null,
+    __userTree: null,
+    __currentUserId: "ODEI-UUID",
+    __selectedLabel: null,
+    __selection: null,
+    __closeBtn: null,
+
+    _createChildControlImpl: function(id) {
+      let control;
+      switch (id) {
+        case "treesLayout":
+          control = new qx.ui.container.Composite(new qx.ui.layout.HBox(10));
+          this._add(control, {
+            flex: 1
+          });
+          break;
+        case "nodeTree":
+        case "userTree":
+          control = new qx.ui.tree.Tree();
+          break;
+        case "selectedFileLayout":
+          control = new qx.ui.container.Composite(new qx.ui.layout.HBox(5)).set({
+            alignY: "middle"
+          });
+          this._add(control);
+          break;
+        case "closeButton":
+          control = new qx.ui.form.Button(this.tr("Close"));
+          this._add(control);
+          break;
+      }
+
+      return control || this.base(arguments, id);
+    },
+
+    __reloadNodeTree: function() {
+      this.__nodeTree.resetRoot();
+
+      const nodeName = this.getNodeModel().getLabel();
+      let root = this.__configureTreeItem(new qx.ui.tree.TreeFolder(), nodeName);
+      root.setOpen(true);
+      this.__nodeTree.setRoot(root);
+
+      // this.__populateNodeFiles();
+      this.__populateFiles(this.__nodeTree, true, false);
+    },
+
+    __reloadUserTree: function() {
+      this.__userTree.resetRoot();
+
+      let root = this.__configureTreeItem(new qx.ui.tree.TreeFolder(), this.__currentUserId);
+      root.setOpen(true);
+      this.__userTree.setRoot(root);
+
+      // this.__populateUserFiles();
+      this.__populateFiles(this.__userTree, false, true);
+    },
+
+    __populateFiles: function(tree, isDraggable = false, isDroppable = false) {
+      const slotName = "listObjects";
+      let socket = qxapp.wrappers.WebSocket.getInstance();
```

Comment:
still using the websocket api ?

Reply:
We still use it for some communication, it will slowly be replaced by the rest api though